### DISCUSSION TITLE
mitmproxy not intercepting certain HTTP/HTTPS requests?

### DISCUSSION THREAD

**Discussion by changsin** [mitmproxy/mitmproxy#6621] [General]
Is there a known case of mitmproxy not intercepting certain requests? We are using mitmproxy to intercept certain web requests (e.g., file uploads), but in one user's machine, we see that it did not catch the requests (should be POST or PUT).  We tried Google Drive, for instance. In other users machines, they seem to work fine so I was wondering if mitmproxy could have some python version compatibility problems. We use pyinstaller to make it one binary so theoretically python compatibility problems should not exist.

**Comment by mhils** [comment]
mitmproxy generally processes all traffic that is sent to it, the only reason why you may not see message bodies is https://docs.mitmproxy.org/stable/concepts-options/#stream_large_bodies. Apart from that it only depends on the proxy mode/client configuration. For example, apps may choose to ignore a configured proxy if they have internet connectivity otherwise.

**Reply by changsin** [reply]
Thanks for the response. After some debugging, we found out the reason for the missing traffic. It turned out that the user was using SwitchyOmega Chrome extension to manage proxy settings on Chrome and thus no Chrome traffic was coming to mitmproxy. Duh! SwitchyOmega's proxy settings were overriding the system proxy settings, in other words.
